cloudwatch_logs_common/src/utils/cloudwatch_facade.cpp

```diff
@@ -147,6 +149,50 @@ Aws::CloudWatchLogs::ROSCloudWatchLogsErrors CloudWatchFacade::CreateLogGroup(
   return status;
 }
 
+Aws::CloudWatchLogs::ROSCloudWatchLogsErrors CloudWatchFacade::CheckLogGroupExists(
+  const std::string & log_group)
+{
+  Aws::CloudWatchLogs::ROSCloudWatchLogsErrors status = CW_LOGS_LOG_GROUP_NOT_FOUND;
+  Aws::CloudWatchLogs::Model::DescribeLogGroupsRequest describe_log_group_request;
+  Aws::String next_token;
+
+  describe_log_group_request.SetLogGroupNamePrefix(log_group.c_str());
+
+  while (CW_LOGS_LOG_GROUP_NOT_FOUND == status) {
+    if (next_token.size() != 0) {
+      describe_log_group_request.SetNextToken(next_token);
+    }
+
+    const auto & response = this->cw_client_.DescribeLogGroups(describe_log_group_request);
+    if (!response.IsSuccess()) {
+      status = CW_LOGS_FAILED;
+      AWS_LOGSTREAM_INFO(__func__, "Request to check if log group named "
```

Comment:
I think this should be a warning, because this is a failure situation but we know how to recover from it

Reply:
done.